packages/airnode-node/src/handlers/start-coordinator.ts

```diff
@@ -88,12 +89,50 @@ function aggregateApiCalls(state: CoordinatorState) {
 async function executeApiCalls(state: CoordinatorState) {
   const { aggregatedApiCallsById } = state;
 
-  const aggregatedApiCalls = Object.values(aggregatedApiCallsById);
+  // TODO: make caching conditional upon endpoint flag "cacheResponses"
+  // https://github.com/api3dao/ois/pull/16
+
+  // TODO: add tests
+
+  const cachedKeys = caching.getKeys('requestId-');
+
+  const filteredUncachedAggregatedApiCalls = pickBy(
+    aggregatedApiCallsById,
+    (value, key) => !cachedKeys.includes(`requestId-${key}`)
+  );
+
+  const filteredCachedAggregatedApiCalls = Object.entries(aggregatedApiCallsById)
+    .filter(([key, _value]) => cachedKeys.includes(`requestId-${key}`))
+    .map(([key, value]) => {
+      const { encodedValue, signature } = caching.getValueForKey(`requestId-${key}`);
+      return {
+        // value.responseValue
+        ...value,
+        // errorMessage: undefined,
+        encodedValue,
+        signature,
+      };
+    });
+
+  const aggregatedApiCalls = Object.values(filteredUncachedAggregatedApiCalls);
+
   const [logs, processedAggregatedApiCalls] = await calls.callApis(aggregatedApiCalls, getWorkerOptions(state));
```

Comment:
We do not want to make the API call for cached requests

Reply:
Right- that's why we now have (emphasis on "Uncached"): 
```ts
const aggregatedApiCalls = Object.values(filteredUncachedAggregatedApiCalls);
```

https://github.com/api3dao/airnode/pull/1372/files#diff-0586244b9496f86cf4c0bdd8ff5bcab4714ed16033fa4f6137e8dacdf732d124R117